lib/src/cmcd/CmcdEncoding.ts

```diff
@@ -1,23 +1,35 @@
+import { CMCD_HEADERS } from './CMCD_HEADERS';
+import { CMCD_JSON } from './CMCD_JSON';
+import { CMCD_QUERY } from './CMCD_QUERY';
+
 /**
  * CMCD encoding types.
  *
  * @group CMCD
  *
+ * @enum
+ *
  * @beta
  */
-export enum CmcdEncoding {
+export const CmcdEncoding = {
 	/**
 	 * JSON
 	 */
-	JSON = 'json',
+	JSON: CMCD_JSON as typeof CMCD_JSON,
```

Comment:
This `as` typing will be needed when `isolateDeclarations` is enabled.

Reply:
I'm not familiar `isolatedDeclarations` yet, but can you double-check what error will occur when you remove `as typeof CMCD_JSON`?

I have a feeling that you don't need `as typeof CMCD_JSON` here, because `CMCD_JSON` will already have the right type. However, I can be totally wrong

Reply:
![Screenshot 2024-10-29 at 07 58 28](https://github.com/user-attachments/assets/d92796da-72be-47db-8480-70200b359394)

Reply:
If you have any other questions about this feature check out [the release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html#isolated-declarations).

Reply:
Thanks, feel free to resolve this conversation!